### PR TITLE
Fix flaky MultiPluginSpec

### PR DESCRIPTION
**Pull request by kpbochenek**

### CODE REVIEW

**Comment by raboof** [comment]
Hmm travis test failures, would be ironic if those are flakey tests again :) - re-triggering them
```

[info] - must find delayed events (2 seconds, 540 milliseconds)
[info] - must find delayed events 2 *** FAILED *** (22 seconds, 543 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (20 seconds) during expectMsg: message matching partial function
[info]   at scala.Predef$.assert(Predef.scala:170)
[info]   at akka.testkit.TestKitBase$class.expectMsgPF(TestKit.scala:404)
[info]   at akka.testkit.TestKit.expectMsgPF(TestKit.scala:828)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectEventWithTimeoutPF(StreamTestKit.scala:589)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNextWithTimeoutPF(StreamTestKit.scala:566)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNextPF(StreamTestKit.scala:558)
[info]   at akka.persistence.cassandra.query.EventsByTagStrictBySeqNoSpec$$anonfun$5$$anonfun$apply$mcV$sp$20.apply(EventsByTagSpec.scala:673)
[info]   at akka.persistence.cassandra.query.EventsByTagStrictBySeqNoSpec$$anonfun$5$$anonfun$apply$mcV$sp$20.apply(EventsByTagSpec.scala:651)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:1078)
[info]   at org.scalatest.TestSuite$class.withFixture(TestSuite.scala:196)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.withFixture(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:1075)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
[info]   at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:1088)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.runTest(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:396)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:373)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:410)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:379)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
[info]   at org.scalatest.WordSpecLike$class.runTests(WordSpecLike.scala:1147)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.runTests(EventsByTagSpec.scala:106)
[info]   at org.scalatest.Suite$class.run(Suite.scala:1147)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.org$scalatest$WordSpecLike$$super$run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
[info]   at org.scalatest.WordSpecLike$class.run(WordSpecLike.scala:1192)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.org$scalatest$BeforeAndAfterAll$$super$run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:210)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:472)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:296)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:286)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]   at java.lang.Thread.run(Thread.java:745)
[info] - must find delayed events 3 *** FAILED *** (22 seconds, 537 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (20 seconds) during expectMsg: message matching partial function
[info]   at scala.Predef$.assert(Predef.scala:170)
[info]   at akka.testkit.TestKitBase$class.expectMsgPF(TestKit.scala:404)
[info]   at akka.testkit.TestKit.expectMsgPF(TestKit.scala:828)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectEventWithTimeoutPF(StreamTestKit.scala:589)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNextWithTimeoutPF(StreamTestKit.scala:566)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNextPF(StreamTestKit.scala:558)
[info]   at akka.persistence.cassandra.query.EventsByTagStrictBySeqNoSpec$$anonfun$5$$anonfun$apply$mcV$sp$21.apply(EventsByTagSpec.scala:704)
[info]   at akka.persistence.cassandra.query.EventsByTagStrictBySeqNoSpec$$anonfun$5$$anonfun$apply$mcV$sp$21.apply(EventsByTagSpec.scala:678)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:1078)
[info]   at org.scalatest.TestSuite$class.withFixture(TestSuite.scala:196)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.withFixture(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:1075)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
[info]   at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:1088)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.runTest(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:396)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:373)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:410)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:379)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
[info]   at org.scalatest.WordSpecLike$class.runTests(WordSpecLike.scala:1147)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.runTests(EventsByTagSpec.scala:106)
[info]   at org.scalatest.Suite$class.run(Suite.scala:1147)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.org$scalatest$WordSpecLike$$super$run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
[info]   at org.scalatest.WordSpecLike$class.run(WordSpecLike.scala:1192)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.org$scalatest$BeforeAndAfterAll$$super$run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:210)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:472)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:296)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:286)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]   at java.lang.Thread.run(Thread.java:745)
[info] - must find delayed events from offset (52 milliseconds)
[info] - must find delayed events when many other events *** FAILED *** (20 seconds, 424 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (20 seconds) during expectMsgClass waiting for class akka.stream.testkit.TestSubscriber$OnNext
[info]   at scala.Predef$.assert(Predef.scala:170)
[info]   at akka.testkit.TestKitBase$class.expectMsgClass_internal(TestKit.scala:486)
[info]   at akka.testkit.TestKitBase$class.expectMsgType(TestKit.scala:459)
[info]   at akka.testkit.TestKit.expectMsgType(TestKit.scala:828)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNextN(StreamTestKit.scala:339)
[info]   at akka.persistence.cassandra.query.EventsByTagStrictBySeqNoSpec$$anonfun$5$$anonfun$apply$mcV$sp$23.apply(EventsByTagSpec.scala:759)
[info]   at akka.persistence.cassandra.query.EventsByTagStrictBySeqNoSpec$$anonfun$5$$anonfun$apply$mcV$sp$23.apply(EventsByTagSpec.scala:739)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:1078)
[info]   at org.scalatest.TestSuite$class.withFixture(TestSuite.scala:196)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.withFixture(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:1075)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
[info]   at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:1088)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.runTest(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:396)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:373)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:410)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:379)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
[info]   at org.scalatest.WordSpecLike$class.runTests(WordSpecLike.scala:1147)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.runTests(EventsByTagSpec.scala:106)
[info]   at org.scalatest.Suite$class.run(Suite.scala:1147)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.org$scalatest$WordSpecLike$$super$run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
[info]   at org.scalatest.WordSpecLike$class.run(WordSpecLike.scala:1192)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.org$scalatest$BeforeAndAfterAll$$super$run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:210)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:472)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:296)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:286)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]   at java.lang.Thread.run(Thread.java:745)
[info] - must find events from many persistenceIds *** FAILED *** (20 seconds, 294 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (20 seconds) during expectMsgClass waiting for class akka.stream.testkit.TestSubscriber$OnNext
[info]   at scala.Predef$.assert(Predef.scala:170)
[info]   at akka.testkit.TestKitBase$class.expectMsgClass_internal(TestKit.scala:486)
[info]   at akka.testkit.TestKitBase$class.expectMsgType(TestKit.scala:459)
[info]   at akka.testkit.TestKit.expectMsgType(TestKit.scala:828)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNextN(StreamTestKit.scala:339)
[info]   at akka.persistence.cassandra.query.EventsByTagStrictBySeqNoSpec$$anonfun$5$$anonfun$apply$mcV$sp$24.apply(EventsByTagSpec.scala:795)
[info]   at akka.persistence.cassandra.query.EventsByTagStrictBySeqNoSpec$$anonfun$5$$anonfun$apply$mcV$sp$24.apply(EventsByTagSpec.scala:775)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:1078)
[info]   at org.scalatest.TestSuite$class.withFixture(TestSuite.scala:196)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.withFixture(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:1075)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
[info]   at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:1088)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.runTest(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:396)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:373)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:410)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
[info]   at scala.collection.immutable.List.foreach(List.scala:381)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:379)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
[info]   at org.scalatest.WordSpecLike$class.runTests(WordSpecLike.scala:1147)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.runTests(EventsByTagSpec.scala:106)
[info]   at org.scalatest.Suite$class.run(Suite.scala:1147)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.org$scalatest$WordSpecLike$$super$run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
[info]   at org.scalatest.WordSpecLike$class.run(WordSpecLike.scala:1192)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.org$scalatest$BeforeAndAfterAll$$super$run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:210)
[info]   at akka.persistence.cassandra.query.AbstractEventsByTagSpec.run(EventsByTagSpec.scala:106)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:472)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:296)
[info]   at sbt.ForkMain$Run$2.call(ForkMain.java:286)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]   at java.lang.Thread.run(Thread.java:745)
```

File: src/test/scala/akka/persistence/cassandra/CassandraLifecycle.scala [hunk 1]
```diff
@@ -27,21 +27,24 @@ object CassandraLifecycle {
...
         system.log.debug("awaitPersistenceInit took {} ms {}", TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - t0), system.name)
       }
     }
   }
 
-  class AwaitPersistenceInit extends PersistentActor {
+  class AwaitPersistenceInit(
+    override val journalPluginId:  String,
+    override val snapshotPluginId: String
+  ) extends PersistentActor {
```
Comment (raboof) [inline src/test/scala/akka/persistence/cassandra/CassandraLifecycle.scala]: I guess this is the core of the PR - can you elaborate a bit on why this change should make the tests less flakey?

Comment (kpbochenek) [inline src/test/scala/akka/persistence/cassandra/CassandraLifecycle.scala]: Right, so previously on start of each test we call 
`CassandraLifecycle.awaitPersistenceInit(system)`
This creates PersistentActor and waits until actor is able to respond with message that received.
And it works fine with all other tests except `MultiPluginSpec` where this functionality was disabled because PersistentActor uses default cassandra-journal unless explicitly said.
And it this suite of tests default cassandra-journal doesn't exist that's why this actor couldn't work here.
But if explicitly said what cassandra-journal actor should operate on we can also use this awaiting step in `MultiPluginSpec` tests.

if `cassandra-journal` / `snapshot-journal` is set to "" then default one is used that's why by default it is empty string

Comment (raboof) [inline src/test/scala/akka/persistence/cassandra/CassandraLifecycle.scala]: Ah, kind of surprising it didn't fail all of the time when referring to a journal that didn't in fact exist.

**Comment by raboof** [comment]
LGTM but it would be helpful if someone more familiar with this plugin could have a second look

File: src/test/scala/akka/persistence/cassandra/journal/MultiPluginSpec.scala [hunk 2]
```diff
@@ -105,6 +105,8 @@ class MultiPluginSpec
 
     session.execute(s"CREATE KEYSPACE IF NOT EXISTS $journalKeyspace WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 }")
     session.execute(s"CREATE KEYSPACE IF NOT EXISTS $snapshotKeyspace WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 }")
+
+    CassandraLifecycle.awaitPersistenceInit(system, "cassandra-journal-a", "cassandra-snapshot-c")
```
Comment (patriknw) [inline src/test/scala/akka/persistence/cassandra/journal/MultiPluginSpec.scala]: looking at the config in this test there are more journals and snapshot-stores, shall we await all of them here?
```
CassandraLifecycle.awaitPersistenceInit(system)
CassandraLifecycle.awaitPersistenceInit(system, "cassandra-journal-a")
CassandraLifecycle.awaitPersistenceInit(system, "cassandra-journal-b")
CassandraLifecycle.awaitPersistenceInit(system, "cassandra-journal-c", "cassandra-snapshot-c")
CassandraLifecycle.awaitPersistenceInit(system, "cassandra-journal-d", "cassandra-snapshot-d")
```

**Comment by patriknw** [comment]
Merging this, I can add the additional await. Thanks a lot @kpbochenek

**Comment by patriknw** [comment]
Refs https://github.com/akka/akka-persistence-cassandra/issues/178
